### PR TITLE
Implement sign in

### PR DESCRIPTION
**Pull request by robinandeer**
## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1444** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1444" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1444?utm_source=stack-comment).

File: apps/store/src/components/BankIdLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { useShopSessionAuthenticateMutation } from '@/services/apollo/generated'
+import { loginMemberSeBankId } from '@/services/authApi/login'
+import { exchangeAuthorizationCode } from '@/services/authApi/oauth'
+import { saveAccessToken } from '@/services/authApi/persist'
+
+export const BankIdLogin = (props: {
+  shopSessionId: string
+  ssn: string
+  onCompleted: () => void
+}) => {
```
Comment (robinandeer) [inline apps/store/src/components/BankIdLogin.tsx:14]: nit: avoid typing inline - I find it harder to read

File: apps/store/src/components/BankIdLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      await authenticateShopSession()
+      props.onCompleted()
+    } catch (error) {
+      datadogLogs.logger.warn('Failed to authenticate', { error })
+      setState('ERROR')
+    }
+  }
+  return (
+    <form onSubmit={handleSubmit}>
+      <Button type="submit" loading={state !== 'IDLE'}>
```
Comment (robinandeer) [inline apps/store/src/components/BankIdLogin.tsx:36]: Button should not be loading if state === ERROR, right? 🤔

File: apps/store/src/components/PriceCalculator/SsnSeSection.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+type Props = {
+  shopSession: ShopSession
+  onCompleted: () => void
+}
+
+// States
+// - Empty input, no customer information
+// - Auth required => Sign in required
+// - Authenticated
```
Comment (robinandeer) [inline apps/store/src/components/PriceCalculator/SsnSeSection.tsx:23]: remove

File: apps/store/src/components/PriceCalculator/SsnSeSection.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    !authenticationStatus ||
+    authenticationStatus === ShopSessionAuthenticationStatus.None
+  ) {
+    return <NewMemberSsnSection shopSession={shopSession} onCompleted={onCompleted} />
+  } else if (authenticationStatus === ShopSessionAuthenticationStatus.AuthenticationRequired) {
+    return <AuthenticationRequiredSsnSection shopSession={shopSession} onCompleted={onCompleted} />
+  } else if (authenticationStatus === ShopSessionAuthenticationStatus.Authenticated) {
+    return <AuthenticatedSsnSection shopSession={shopSession} onCompleted={onCompleted} />
+  } else {
+    const status: never = authenticationStatus
```
Comment (robinandeer) [inline apps/store/src/components/PriceCalculator/SsnSeSection.tsx:38]: never seen this pattern before -- why "never"?

File: apps/store/src/components/PriceCalculator/SsnSeSection.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    </>
+  )
+}
+SsnSeSection.sectionId = 'ssn-se'
+
+const AuthenticationRequiredSsnSection = ({ shopSession, onCompleted }: Props) => {
+  const { ssn } = shopSession.customer ?? {}
+  if (!ssn) {
+    throw new Error('Must have ssn at this point')
+  }
```
Comment (robinandeer) [inline apps/store/src/components/PriceCalculator/SsnSeSection.tsx:92]: why not pass in SSN as prop instead?

**Review by robinandeer — COMMENTED** [review]
Left some comments -- feel free to address later though and merge this now

**Review by alebedev — APPROVED** [review]
Pair-reviewed with Robin, fixed everything, good to merge

**Comment by alebedev** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: KjKPf771z3voCWBQ1Jh5.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1444)

**Pull request merged**